tests/unit/test_router.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from gordon import router
+
+
+@pytest.fixture()
+def router_inst(inited_plugins, plugin_kwargs):
+    inited_plugins.pop('runnable')
+    router_inst = router.GordonRouter(plugins=inited_plugins, **plugin_kwargs)
+    return router_inst
+
+
+class EventMsgStub:
```

Comment:
Is there a way [/any reason not] to explicitly use the interface in the class definition?  I realize I am missing some context on zope, and anyways we're not using it yet, but I thought we had mentioned somewhere that there is a way to enforce or assert that all required methods are implemented?

The thing I'm concerned about is if in the future we change the IEventMessage, but forget to update some routing method, our tests won't catch it because they're only testing against a Stub that maybe has nothing to do with the current state of the interface anymore.

Reply:
Are you meaning something [like this](https://github.com/spotify/gordon/blob/develop/tests/unit/conftest.py#L38)? If so, then I just forgot to add that decorator to this. But it's also not _really_ needed because I don't do anything w/i the router that checks to see if the event message provided is in fact a provider for the interface. Typing that out makes me feel like I should somewhere, though... and agreeing with what you're saying w/r/t testing. I'll add that.

FWIW The functionality to assert a correct implementation isn't in gordon (or gordon janitor) yet (there's an internal ticket for that). The functionality to find out who _says_ they've implemented an interface exists in gordon/main.py - but anyone can declare they implement something with that aforementioned decorator. Not sure if that makes sense.

Reply:
Yep, that was the thing I meant!  I realize it doesn't do much yet, so its not a blocker for me to skip it.

Reply:
Hey I just pushed another commit - I'll squash this if it looks good to you - that basically ignores event messages that do not "provide" the `IEventMessage` interface (to be switched to "verify" once we work on that ability).